### ISSUE TITLE
[a11y]: UI Shell - HeaderPanel not accessible in Safari

### ISSUE THREAD

**Issue by heloiselui** [carbon-design-system/carbon#19771]
### Package

@carbon/react

### Browser

Safari

### Operating System

MacOS

### Package version

1.85.1

### Automated testing tool and ruleset

Manual QA

### Description

In Safari, the UI Shell components "Header w/ Actions and Right Panel" and "Header w/ Actions and Switcher" do not handle keyboard navigation properly. 
This seems to be related to the onBlur and focus event timing differences in Safari. (not sure)

Expected behavior ([Carbon documentation](https://carbondesignsystem.com/components/UI-shell-right-panel/accessibility/)):
- User can open the Switcher with keyboard (Enter/Space)
- User can tab through items
- User can close by clicking outside

In the video below, I'm trying to tab through the elements:

https://github.com/user-attachments/assets/dcca1abc-419f-41ee-bab2-ebf7089b709f

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-actions-and-right-panel

### Steps to reproduce

- Open the storybook link in Safari
- Go to the `UI Shell` > `Header w/ Actions and Right Panel `| `Header w/ Actions and Switcher`:
  - Try to open the Switcher with keyboard (Enter/Space)
  - Try to tab through items
  - Try to close by clicking outside

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
